src/services/jobProcessor.ts

```diff
@@ -0,0 +1,63 @@
+import { setTimeout as setTimeoutPromise } from 'timers/promises';
+import { Logger } from '@map-colonies/js-logger';
+import { inject, injectable } from 'tsyringe';
+import { IJobResponse } from '@map-colonies/mc-priority-queue';
+import { QueueClient } from '../clients/queueClient';
+import { SERVICES } from '../common/constants';
+import { IConfig, IQueueConfig, LogContext } from '../common/interfaces';
+import { JOB_HANDLER_FACTORY_SYMBOL, JobHandlerFactory } from './jobHandlerFactory';
+
+@injectable()
+export class JobProcessor {
+  private readonly logContext: LogContext;
+  private readonly jobTypes;
+  private readonly initTaskType;
+  private readonly dequeueIntervalMs: number;
+  private readonly isRunning = true;
+  public constructor(
+    @inject(SERVICES.LOGGER) private readonly logger: Logger,
+    @inject(SERVICES.CONFIG) private readonly config: IConfig,
+    @inject(JOB_HANDLER_FACTORY_SYMBOL) private readonly jobHandlerFactory: JobHandlerFactory,
+    private readonly queueClient: QueueClient
+  ) {
+    const { jobTypes, initTaskType, dequeueIntervalMs } = this.config.get<IQueueConfig>('queue');
+    this.dequeueIntervalMs = dequeueIntervalMs;
+    this.jobTypes = jobTypes;
+    this.initTaskType = initTaskType;
+    this.logContext = {
+      fileName: __filename,
+      class: JobProcessor.name,
+    };
+  }
+
+  public async consumeAndProcess(): Promise<void> {
+    const logCtx: LogContext = { ...this.logContext, function: this.consumeAndProcess.name };
+    try {
+      for await (const job of this.getJob()) {
+        const jobHandler = this.jobHandlerFactory(job.type);
```

Comment:
jobHandlerFactory assumes confugured job type is same as hard coded jobTypes as defined in constants file -this may lead to bugs

Reply:
now driven from the config